Sally7/S7Connection.cs

```diff
@@ -69,22 +75,12 @@ public S7Connection(string host, Tsap sourceTsap, Tsap destinationTsap, MemoryPo
         }
 
         /// <summary>
-        /// Gets or sets the ReceiveTimeout of the underlying <see cref="TcpClient"/>.
+        /// Gets or sets the timeout, in milliseconds, for performing requests.
         /// </summary>
-        public int ReceiveTimeout
-        {
-            get => TcpClient.ReceiveTimeout;
-            set => TcpClient.ReceiveTimeout = value;
-        }
-
-        /// <summary>
-        /// Gets or sets the SendTimeout of the underlying <see cref="TcpClient"/>.
-        /// </summary>
-        public int SendTimeout
-        {
-            get => TcpClient.SendTimeout;
-            set => TcpClient.SendTimeout = value;
-        }
+        /// <remarks>
+        /// The default value is <see cref="DefaultRequestTimeout"/>.
+        /// </remarks>
+        public int RequestTimeout { get; set; } = DefaultRequestTimeout;
```

Comment:
Right, but the below -1ms restriction applies to the int version just as well.

My experience with writing code is that using strong types when units are involved is nearly always better, in particular for time durations. I'd much rather use a dedicated duration type rather than a primitive type without a unit, even if it doesn't perfectly match some other API.

You could document the limitations of the `CancelAfter` or just let the `ArgumentOutOfRangeException` take care of that and don't bother - which is good enough in my book.

Reply:
> My experience with writing code is that using strong types when units are involved is nearly always better, in particular for time durations. I'd much rather use a dedicated duration type rather than a primitive type without a unit, even if it doesn't perfectly match some other API.

You're absolutely right. Let's not dwell on why Microsoft chose to use `int` originally, the right thing to do is `TimeSpan`.

> You could document the limitations of the CancelAfter or just let the ArgumentOutOfRangeException take care of that and don't bother - which is good enough in my book.

I'll document and validate, because the exception from `CancelAfter` will only show when sending a request, not when setting the timeout.